src/lein_monolith/dependency.clj

```diff
@@ -165,6 +165,74 @@
           (recur (conj result node)
                  (into (pop queue) (set/difference consumers result))))))))
 
+(defn unique-cycles
+  "Return a set of all unique cycles in dependency graph m."
+  [m]
+  {:pre [(map? m)]
+   :post [(set? %)]}
+  (let [path->cycles (fn path->cycles [path]
+                       {:pre [(seq path)]}
+                       (let [k (peek path)
+                             mks (set (keys m))
+                             vs (m k)
+                             path-set (set path)]
+                         (mapcat (fn [v]
+                                   (cond
+                                     ; found the cycle
+                                     (path-set v) [(into []
+                                                         ; drop non-cyclic prefix
+                                                         (drop-while (complement #{v}))
+                                                         (conj path v))]
+                                     :else (path->cycles
+                                             (conj path v))))
+                                 vs)))
+        all-cycles (mapcat #(path->cycles [%]) (keys m))
+        ; remove duplicate cycles (that involve the same deps)
+        [cycles-vecs _] (reduce (fn [[cycles-vecs cycle-sets] c]
+                                  (let [cset (set c)]
+                                    (if (cycle-sets cset)
+                                      [cycles-vecs cycle-sets]
+                                      [(conj cycles-vecs c)
+                                       (conj cycle-sets cset)])))
+                                [#{} #{}]
+                                all-cycles)]
+    cycles-vecs))
+
+
+(defn pretty-cycle
+  "Returns a pretty-printed string representation of cycle c.
+  
+  eg. (println (pretty-cycle [1 2 3 1])) =>
+      + 1
+      ^ + 2
+      |  + 3
+      |_/"
+  [c]
+  {:pre [(vector? c)
+         (apply = ((juxt first peek) c))]}
+  (if (= 2 (count c))
+    (str/join \newline
+              [(str "+ " (pr-str (peek c)))
+               "^\\"
+               "|_|"])
+    (str/join (map-indexed (fn [indent el]
+                             (condp = indent
```

Comment:
Similar to the above, this `condp` only has one test clause plus the default, so it should be an `if`.